packages/client/src/pages/self_register/index.tsx

```diff
@@ -68,6 +68,7 @@ const SelfRegisterPage: React.FC = () => {
       setErrors({
         registrationCode: t(ValidationMessage.InvalidRegistrationCode),
       });
+      setSubmitting(false);
     }
```

Comment:
Ok, so I would wrap this into a try { ...all of the code } finally { setSubmitting(false) }. That way, whatever happens, the submission will be done (as far as the submitting state is concerned)